chatgpt/nodes/validator.py

```diff
@@ -162,16 +179,25 @@ def score_validator_performance(self, own_submission: Dict[str, Any], validator_
 
         scores = {}
 
-        for dimension in ['overall_score', 'authenticity', 'ownership', 'quality', 'uniqueness']:
+        for dimension in ['score', 'authenticity', 'ownership', 'quality', 'uniqueness']:
             if dimension in own_submission and dimension in validator_score:
-                scores[dimension] = 1 - abs(own_submission[dimension] - validator_score[dimension]) / 100
+                scores[dimension] = 1 - abs(own_submission[dimension] - validator_score[dimension] / 1e18)
             else:
                 scores[dimension] = 0 if dimension in own_submission else 1
 
-        file_added_block = file_data['addedAtBlock']
-        max_block_difference = self.config.node.max_wait_blocks
-        block_difference = validator_score['reportedAtBlock'] - file_added_block
-        scores['speed'] = max(0, 1 - (block_difference / max_block_difference))
+        if 'addedAtBlock' in file_data and 'reportedAtBlock' in validator_score:
+            file_added_block = file_data['addedAtBlock']
+            reported_block = validator_score['reportedAtBlock']
+            max_block_difference = self.config.node.max_wait_blocks
+
+            if reported_block >= file_added_block:
+                block_difference = reported_block - file_added_block
+                scores['speed'] = max(0, 1 - (block_difference / max_block_difference))
+            else:
+                # This case shouldn't occur in normal circumstances
+                scores['speed'] = 0
+        else:
```

Comment:
This case is not needed, max gives 0 in this case.

Reply:
I think this case is only hit if file_data.addedAtBlock or validator_score.reportedAtBlock doesn't exist? In which case, the max won't be executed.

Reply:
Sorry, I meant to add that comment to the inner else three lines up

Reply:
Max won't be executed if `reported_block < file_added_block`, which is the unusual case this `else` handles. I will just initialize `speed` to 0.